manager/sql/manager_sql.go

```diff
@@ -237,27 +129,41 @@ func (s *SQLManager) create(policy Policy, tx *sqlx.Tx) (err error) {
 		}
 	}
 
-	switch s.db.DriverName() {
-	case "postgres", "pgx":
-		if _, err = tx.Exec(s.db.Rebind("INSERT INTO ladon_policy (id, description, effect, conditions) SELECT ?, ?, ?, ? WHERE NOT EXISTS (SELECT 1 FROM ladon_policy WHERE id = ?)"), policy.GetID(), policy.GetDescription(), policy.GetEffect(), conditions, policy.GetID()); err != nil {
-			return errors.WithStack(err)
-		}
-	case "mysql":
-		if _, err = tx.Exec(s.db.Rebind("INSERT IGNORE INTO ladon_policy (id, description, effect, conditions) VALUES (?, ?, ?, ?)"), policy.GetID(), policy.GetDescription(), policy.GetEffect(), conditions); err != nil {
-			return errors.WithStack(err)
-		}
-	default:
-		return errors.Errorf("Database driver %s is not supported", s.db.DriverName())
+	if _, ok := Databases[s.database]; !ok {
+		return errors.Errorf("Database %s is not supported", s.database)
+	}
+
+	if _, err = tx.Exec(Databases[s.database].CreatePolicy, policy.GetID(), policy.GetDescription(), policy.GetEffect(), conditions); err != nil {
+		return errors.WithStack(err)
 	}
 
 	type relation struct {
 		p []string
 		t string
 	}
-	var relations = []relation{{p: policy.GetActions(), t: "action"}, {p: policy.GetResources(), t: "resource"}, {p: policy.GetSubjects(), t: "subject"}}
+	var relations = []relation{
+		{p: policy.GetActions(), t: "action"},
+		{p: policy.GetResources(), t: "resource"},
+		{p: policy.GetSubjects(), t: "subject"},
+	}
 
-	for _, v := range relations {
-		for _, template := range v.p {
+	for _, rel := range relations {
+		var subject string
```

Comment:
This should be query or something, subject implies that it's a policy subject which isn't the case

Reply:
Agreed. I struggled to find the right variable name for this, and `query` still doesn't seem quite right, but it is at least less confusing than `subject`.